test/new-e2e/ndm/snmp/snmpTestEnv.go

```diff
@@ -0,0 +1,132 @@
+package snmp
+
+import (
+	"context"
+	"embed"
+	"errors"
+	"fmt"
+	"path"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/utils/infra"
+	"github.com/DataDog/test-infra-definitions/aws/ec2/ec2"
+	"github.com/DataDog/test-infra-definitions/datadog/agent"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+)
+
+type TestEnv struct {
+	InstanceIP  string
+	StackOutput auto.UpResult
+}
+
+//go:embed compose/snmpCompose.yaml
+var snmpCompose string
+
+//go:embed config/public.yaml
+var snmpConfig string
+
+const (
+	composeDataPath = "compose/data"
+)
+
+func NewTestEnv(name, keyPairName, ddAPIKey, ddAPPKey string, shouldDestroy bool) (*TestEnv, error) {
+	ctx := context.Background()
+	snmpTestEnv := &TestEnv{}
+
+	stackManager := infra.GetStackManager()
+	envName := "aws/sandbox"
+	instanceName := fmt.Sprintf("snmp-agent-%s", name)
+
+	if shouldDestroy {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏It feels a weird UX to me that NewTestEnv either creates a stack and run test or delete an existing stack. Should it be two separate commands/entrypoints? For instance `ddev` has different entrypoints to start a new env, test on an existing env and delete an existing env. It also provides a convenient entrypoint for doing all three in a row

Reply:
Changed it ! Moreover, I realised you need to first upsert a stack before deleting it - the stack manager needs to load the stack state syncing with remote